pkg/awsclient/client.go

```diff
@@ -376,15 +407,22 @@ func newClient(controllerName, awsAccessID, awsAccessSecret, token, region strin
 	if controllerName != "" {
 		// NOTE(efried): We can't count on the returned client being used by a single thread, so we
 		// create a separate timer for each request.
-		timers := make(map[*request.Request]time.Time)
+		timers := &timerMap{
+			theMap: make(map[string]time.Time),
+		}
 		// Validate is the first phase of a request. Start the timer as early as possible (as the first handler there).
 		s.Handlers.Validate.PushFront(func(r *request.Request) {
-			timers[r] = time.Now()
+			timers.start(r.RequestID)
 		})
 		// Complete is the last phase of a request. Stop the timer as late as possible (as the last handler there).
 		s.Handlers.Complete.PushBack(func(r *request.Request) {
-			defer func() { delete(timers, r) }()
-			localmetrics.Collector.AddAPICall(controllerName, r.HTTPRequest, r.HTTPResponse, time.Since(timers[r]).Seconds())
+			dur := timers.end(r.RequestID)
+			if dur > 0 {
+				localmetrics.Collector.AddAPICall(controllerName, r.HTTPRequest, r.HTTPResponse, dur)
+			} else {
+				localmetrics.Collector.AddAPICall(controllerName, r.HTTPRequest, r.HTTPResponse, 123456)
+
```

Comment:
Not actually going to do this, just using it as an eyecatcher so I can tell whether the RequestID is working (it isn't). We should really have logging here, but that'll take some additional work.